tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc

```diff
@@ -2126,6 +2162,120 @@ TEST_P(OpConverter_FP32_Test, ConvertTranspose) {
+
+  struct TileParam {
+    std::vector<int> input_dims;
+    std::vector<int> multiplier;
+    std::vector<float> tensor;
+    // Concrete (static) output dimensions, including batch size as first dim
+    std::vector<int> expected_output_dims;
+    std::vector<int> expected_results;
+    // Expected status of conversion (with concrete error message)
+    Status status;
+    // Expected status of BuildAndRun
+    Status runtime_status;
+  };
+
+  std::vector<TileParam> test_params = {
+      TileParam{{1, 2, 3},  // input dims
+                {1, -2, 1}, // multiplier
+                {},
+                {},         // output dims
+                {},
+                 Status(error::INVALID_ARGUMENT,
+                        "All replications of the Tile operation in "
+                        "my_tile should be positive, got (1, -2, 1)")},
+      TileParam{{1, 2, 3},     // input dims
+                {1, 2, 1, 3},  // multiplier
+                {0, 1, 2, 3, 4, 5},
+                {},            // output dims
+                {},
+                 Status(error::INVALID_ARGUMENT,
+                        "The lenght of the replication vector (4) of the Tile operation in 'my_tile' "
+                        "is expected to be equal to the number of dimentions of the input vector (3)")},
+      TileParam{{1, 2},     // input dims
+                {1, 3},     // multiplier
+                {2, 3},     // input values
+                {1, 6},     // output dims
+                {2, 3, 2, 3, 2, 3}},  // out values
+      TileParam{{1, 2, 3},  // input dims
+                {1, 2, 1},  // multiplier
+                {0, 1, 2, 3, 4, 5},
+                {1, 4, 3},  // output dims
+                {0, 1, 2, 3, 4, 5,
+                 0, 1, 2, 3, 4, 5}},
+      TileParam{{1, 2, 3},  // input dims
+                {1, 1, 2},  // multiplier
+                {0, 1, 2, 3, 4, 5},
+                {1, 2, 6},  // output dims
+                {0, 1, 2, 0, 1, 2, 3, 4, 5, 3, 4, 5}},
+      TileParam{{1, 2, 3},  // input dims
+                {1, 2, 2},  // multiplier
+                {0, 1, 2, 3, 4, 5},
+                {1, 4, 6},  // output dims
+                {0, 1, 2, 0, 1, 2, 3, 4, 5, 3, 4, 5,
+                 0, 1, 2, 0, 1, 2, 3, 4, 5, 3, 4, 5}}
+  };
+
+  for (bool multiplier_is_tensor : {true, false}) {
+    for (bool input_is_tensor : {true, false}) {
+      int k = 0;
+      for (auto p : test_params) {
+        ++k;
+        std::vector<int>num_mults = {p.multiplier.size()};
+        std::vector<std::vector<int>>partial_input_dims_options = {{}};
+        if (multiplier_is_tensor) {
+          if (trt_mode_ == TrtTestMode::kImplicitBatch) {
+             p.status = Status(error::INVALID_ARGUMENT,
+                             "Conversion for Tile is not implementd for multipliers passed as a tensor");
+             num_mults = {1, p.multiplier.size()};
+          } else {
+            if (k <= 2) {
```

Comment:
The `k <= 2` / `k > 2` condition here seems pretty brittle. I think you are basing this on whether the `p.status.ok()` is true or false...

Reply:
I didn't like it either. Improved.